### PR TITLE
Resolve issue with IS DISTINCT FROM and Decimal values with precision > 18

### PR DESCRIPTION
**Pull request by ptkool**
Correct the logic in method `distinctBlockPositionLongLong` to get the proper values for `rightLow` and `rightHigh`.

This PR resolve issue https://github.com/prestosql/presto/issues/1984.

### CODE REVIEW

**Comment by martint** [comment]
Wow, good catch!

You can add the test to io.prestosql.type.TestDecimalOperators

**Comment by ptkool** [comment]
@martint I've added a test in `io.prestosql.block.TestInt128ArrayBlock`. The problem can't be reproduced with a test in `TestDecimalOperators`. It's a situation where decimal values are being read from a table and Presto creates instances of `Int128ArrayBlock` to represent these values.

File: presto-main/src/test/java/io/prestosql/block/TestInt128ArrayBlock.java [hunk 1]
```diff
@@ -82,6 +85,19 @@ public void testCompactBlock()
         testIncompactBlock(new Int128ArrayBlock(valueIsNull.length - 2, Optional.of(valueIsNull), longArray));
     }
 
+    @Test
+    public void testIsDistinctFrom()
+    {
+        long[] arr1 = {112L, 0L}; // DECIMAL '1.12'
```
Comment (martint) [inline presto-main/src/test/java/io/prestosql/block/TestInt128ArrayBlock.java]: These can be inlined. There's no real benefit in having them as standalone variables.

Comment (ptkool) [reply presto-main/src/test/java/io/prestosql/block/TestInt128ArrayBlock.java]: I'll take care of it.

**Comment by martint** [comment]
Merged, thanks for your contribution!

**Pull request closed without merging**